File: Temp/leetcode/editor/cn/[1014]最佳观光组合.c
// 1014 最佳观光组合
// https://leetcode.cn/problems/best-sightseeing-pair/


//leetcode submit region begin(Prohibit modification and deletion)
int maxScoreSightseeingPair(int* values, int valuesSize) {
    int ans = 0, mx = 0;
    for (int i = 0; i < valuesSize; i++) {
        int v = values[i];
        if (v - i + mx > ans)
            ans = v - i + mx;
        if (v + i > mx)
            mx = v + i;
    }
    return ans;
}
//leetcode submit region end(Prohibit modification and deletion)

File: Temp/leetcode/editor/cn/[1014]最佳观光组合.py
# 1014 最佳观光组合
# https://leetcode.cn/problems/best-sightseeing-pair/
from imports import *


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def maxScoreSightseeingPair(self, values: List[int]) -> int:
        # vi + vj + i - j
        # (vi + i) + (vj - j)
        # 遍历 vj （j在右边）更新答案
        # 计算 vj + j 更新 mx
        ans = mx = 0
        for i, v in enumerate(values):
            ans = max(ans, v - i + mx)
            mx = max(mx, v + i)
        return ans
# leetcode submit region end(Prohibit modification and deletion)
